dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -661,6 +663,78 @@ def test_sample_size_passed_to_profile(self, *mocks):
         profiler._sampling_ratio = 0.2
         self.assertEqual(10000, update_mock.call_args[0][1])
 
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnPrimitiveTypeProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnStatsProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    def test_index_overlap_resolved_for_update_profile(self, *mocks):
+        data = pd.Series([0, None, 1, 2, None])
+        profile = StructuredDataProfile(data)
+        self.assertEqual(0, profile.min_id)
+        self.assertEqual(4, profile.max_id)
+        self.assertDictEqual(profile.null_types_index, {'nan': {1, 4}})
+        profile.update_profile(data)
+        # Now all indices will be shifted by max_id + 1 (5)
+        # So the 2 None will move from indices 1, 4 to 6, 9
+        self.assertEqual(0, profile.min_id)
+        self.assertEqual(9, profile.max_id)
+        self.assertDictEqual(profile.null_types_index, {'nan': {1, 4, 6, 9}})
+
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnPrimitiveTypeProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnStatsProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    def test_index_overlap_resolved_for_merge(self, *mocks):
+        data = pd.Series([0, None, 1, 2, None])
+        profile1 = StructuredDataProfile(data)
+        profile2 = StructuredDataProfile(data)
+
+        # Ensure merged profile included shifted indices
+        profile3 = profile1 + profile2
+        self.assertEqual(0, profile3.min_id)
+        self.assertEqual(9, profile3.max_id)
+        self.assertDictEqual(profile3.null_types_index, {'nan': {1, 4, 6, 9}})
+
+        # Ensure original profile not overwritten
+        self.assertEqual(0, profile2.min_id)
+        self.assertEqual(4, profile2.max_id)
+        self.assertDictEqual(profile2.null_types_index, {'nan': {1, 4}})
```

Comment:
can we add this check for profile1 as well?

Reply:
I figured it was unnecessary since the test function before this one uses the same data and asserts it about "profile1" in that function, but I can add it back.